src/open_clip/loss.py

```diff
@@ -99,33 +99,44 @@ def get_ground_truth(self, device, num_logits) -> torch.Tensor:
             labels = self.labels[device]
         return labels
 
-    def get_logits(self, image_features, text_features, logit_scale):
+    def get_logits(self, features_a, features_b, logit_scale):
+
         if self.world_size > 1:
-            all_image_features, all_text_features = gather_features(
-                image_features, text_features,
+            all_features_a, all_features_b = gather_features(
+                features_a, features_b,
                 self.local_loss, self.gather_with_grad, self.rank, self.world_size, self.use_horovod)
 
             if self.local_loss:
-                logits_per_image = logit_scale * image_features @ all_text_features.T
-                logits_per_text = logit_scale * text_features @ all_image_features.T
+                logits_per_feature_a = logit_scale * features_a @ all_features_b.T
+                logits_per_feature_b = logit_scale * features_b @ all_features_a.T
             else:
-                logits_per_image = logit_scale * all_image_features @ all_text_features.T
-                logits_per_text = logits_per_image.T
+                logits_per_feature_a = logit_scale * all_features_a @ all_features_b.T
+                logits_per_feature_b = logits_per_feature_a.T
         else:
-            logits_per_image = logit_scale * image_features @ text_features.T
-            logits_per_text = logit_scale * text_features @ image_features.T
+            logits_per_feature_a = logit_scale * features_a @ features_b.T
+            logits_per_feature_b = logit_scale * features_b @ features_a.T
+
+       
+        return logits_per_feature_a, logits_per_feature_b
+
+
+    def forward(self, image_features=None, text_features=None, logit_scale=None, text_a_features=None, text_b_features=None, output_dict=False):
         
-        return logits_per_image, logits_per_text
+        if image_features is not None and text_features is not None:
```

Comment:
This section needs a check for the case of both image/text are none, since that's something new that can occur now I think?